ui/apps/ui/src/app/pages/guidelines-page/guideline-detail-page.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, OnInit } from '@angular/core';
+import { IResult } from '@collections/repositories/types';
+import { GuidelinesService } from './guidelines.service';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { guidelinesAdapter } from '@collections/data/guidelines/adapter.data';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import moment from 'moment';
+import { Observable, catchError, map, of, switchMap } from 'rxjs';
+import { IGuideline } from '@collections/data/guidelines/guideline.model';
+
+interface IIGDictionary {
+  [key: string]: string;
+}
+
+@UntilDestroy()
+@Component({
+  selector: 'ess-guideline-detail-page',
+  templateUrl: './guideline-detail-page.component.html',
+  styleUrls: ['./guideline-detail-page.component.scss'],
+})
+export class GuidelineDetailPageComponent implements OnInit {
+  guideline?: IResult;
+  interoperabilityGuidelineItem?: IGuideline;
+  currentTab = 'about';
+
+  interoperabilityGuidelineStatusesDictionary: IIGDictionary = {
+    'ir_status-abandoned': 'Abandoned',
+    'ir_status-accepted': 'Accepted',
+    'ir_status-candidate': 'Candidate',
+    'ir_status-proposed': 'Proposed',
+    'ir_status-consultation': 'Consultation',
+    'ir_status-deprecated': 'Deprecated',
+    'ir_status-on_hold': 'On Hold',
+    'ir_status-operating': 'Operating',
+    'ir_status-rejected': 'Rejected',
+    'ir_status-withdrawn': 'Withdrawn',
+  };
+
+  interoperabilityGuidelineTypeDictionary: IIGDictionary = {
+    'ir_eosc_guideline_type-eosc_core_interoperability_guideline':
+      'EOSC-Core Interoperability Guideline',
+    'ir_eosc_guideline_type-eosc_exchange_interoperability_guideline_horizontal':
+      'EOSC-Exchange Interoperability Guideline (Horizontal)',
+    'ir_eosc_guideline_type-eosc_exchange_interoperability_guideline_thematic':
+      'EOSC-Exchange Interoperability Guideline (Thematic)',
+    'ir_eosc_guideline_type-operational_baseline': 'Operational Baseline',
+  };
+
+  guidelineId$: Observable<IGuideline | null> =
+    new Observable<IGuideline | null>();
+
+  constructor(
+    private guidelinesService: GuidelinesService,
+    private route: ActivatedRoute,
+    private _router: Router
+  ) {}
+
+  ngOnInit(): void {
+    this.guidelineId$ = this.route.params.pipe(
+      map((params: Params) => params['guidelineId']),
+      switchMap((param) => {
+        return this.guidelinesService.get$(param).pipe(
+          catchError((error) => {
+            this._router.navigate(['**']);
```

Comment:
this should be returned as an observable - doing asynchronous task within observable pipeline can lead to multiple issues.

```
return `from(this._router.navigate(["**"])).pipe(map(() => NEVER))`
```

or something like this could be used.